lib/rules/jsx-no-useless-fragment.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * @fileoverview Disallow useless fragments
+ */
+
+'use strict';
+const docsUrl = require('../util/docsUrl');
+
+/**
+ * Test whether a JSXElement is a fragment
+ * @param {JSXElement} node
+ * @returns {boolean}
+ */
+function isFragment(node) {
+  const name = node.openingElement.name;
+
+  // <Fragment>
+  if (
+    name.type === 'JSXIdentifier'
+    && name.name === 'Fragment'
+  ) {
+    return true;
+  }
+
+  // <*.Fragment>
```

Comment:
we shouldn't check anything dot Fragment; we should follow the pragma settings here.